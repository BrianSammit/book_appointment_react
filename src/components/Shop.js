import React, { Component } from 'react';
import Countdown from './Countdown';

class Shop extends Component {
  constructor(props) {
    super(props);

    this.state = {
      countdown: {
        futureDate: '2022-12-31 00:00:00',
      },
    };
  }

  render() {
    const { countdown } = this.state;
    return (
      <div className="shop-container">
        <div className="shop-square">
          <Countdown futureDate={countdown.futureDate}></Countdown>
        </div>
      </div>
    );
  }
}

export default Shop;
