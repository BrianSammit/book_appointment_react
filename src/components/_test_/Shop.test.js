/**
 * @jest-environment jsdom
 */

import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Shop from '../Shop';

describe('Creates a component', () => {
  it('Renders a component', () => {
    render(
      <Router>
        <Shop />
      </Router>,
    );
    const element = screen.getByText('COMMING SOON !!');
    expect(element).toBeInTheDocument();
  });
});
