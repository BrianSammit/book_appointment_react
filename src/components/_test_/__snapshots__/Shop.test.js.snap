// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a component renders correctly 1`] = `
<div
  className="shop-container"
>
  <div
    className="shop-square"
  >
    <div
      className="countdown"
    >
      <div
        className="countdown-segment"
      >
        <span
          className="countdown-segment-number"
        >
          429
        </span>
        <span
          className="countdown-segment-caption"
        >
          DAYS
        </span>
      </div>
      <div
        className="countdown-segment"
      >
        <span
          className="countdown-segment-number"
        >
          16
        </span>
        <span
          className="countdown-segment-caption"
        >
          HOURS
        </span>
      </div>
      <div
        className="countdown-segment"
      >
        <span
          className="countdown-segment-number"
        >
          0
        </span>
        <span
          className="countdown-segment-caption"
        >
          MIN
        </span>
      </div>
      <div
        className="countdown-segment"
      >
        <span
          className="countdown-segment-number"
        >
          28
        </span>
        <span
          className="countdown-segment-caption"
        >
          SEC
        </span>
      </div>
    </div>
    <h2
      className="comming"
    >
       COMMING SOON !!
    </h2>
  </div>
</div>
`;
